src/syntax/ast.rs

```diff
@@ -28,7 +31,6 @@ pub struct Pattern {
 
 #[derive(Debug, PartialEq)]
 pub enum Expression {
-    Pattern(Pattern),
```

Comment:
I'm wondering if instead we should have:
```
pub struct Pattern {
  pub elements: Vec<PatternElement>
}

pub enum PatternElement {
  TextElement(String),
  Expression(Expression)
}
```

Because adding `PatternElement` into `Expression seems to be useless as those elements can be anywhere except of inside Pattern.elements.